### PR TITLE
misc: Add DRAMSys tests to our weekly tests

### PR DESCRIPTION
**Pull request by mkjost0**
This adds the DRAMSys tests to our weekly-tests.yaml file

Change-Id: Ieb7903a3a7ffae6359b3de5f66e1dd65eb51fc80

### CODE REVIEW

**Comment by BobbyRBruce** [comment]
The issue is you aren't recursively getting the submodules. The README file for DRAMSys was incorrect. I fix this in #202. Update the test to incorporate this change and it should start working.

**Comment by mkjost0** [comment]
@BobbyRBruce I tested that out and the test runs successfully now, thanks! I'll make the change here then mark it as ready for review

File: .github/workflows/weekly-tests.yaml [hunk 1]
```diff
@@ -77,3 +77,34 @@ jobs:
...
+        run: |
+          git clone https://github.com/tukl-msd/DRAMSys DRAMSys
+          cd DRAMSys
+          git checkout -b gem5 09f6dcbb91351e6ee7cadfc7bc8b29d97625db8f
+          git submodule update --init --recursive
+
+      # gem5 is built separately because it depends on the DRAMSys library
+      - name: Build gem5
+        working-directory: ${{ github.workspace }}
+        run: scons build/ALL/gem5.opt -j4
```
Comment (BobbyRBruce) [inline .github/workflows/weekly-tests.yaml]: I'm not sure how we do this elsewhere in out workflow files but shouldn't we use something like `nproc` here?

**Review by BobbyRBruce — CHANGES_REQUESTED** [review]

File: .github/workflows/weekly-tests.yaml (hunk 1 above)
Comment (mkjost0) [reply .github/workflows/weekly-tests.yaml]: Ahh, yeah we do, I can update that! I think it was something I forgot to change when I was testing

**Review by BobbyRBruce — APPROVED** [review]

**Pull request merged**